src/GovUk.Education.ExploreEducationStatistics.Admin.Tests/Services/ContentServiceTests.cs

```diff
@@ -169,17 +172,317 @@ await using (var contentDbContext = InMemoryApplicationDbContext(contentDbContex
+                CreatedById = Guid.NewGuid(),
+            };
+            var release = new Release
+            {
+                Content = new List<ReleaseContentSection>
+                {
+                    new ReleaseContentSection
+                    {
+                        ContentSection = new ContentSection
+                        {
+                            Content = new List<ContentBlock>
+                            {
+                                new HtmlBlock
+                                {
+                                    Comments = new List<Comment>
+                                    {
+                                        comment
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+            };
+
+            var contentDbContextId = Guid.NewGuid().ToString();
+            await using (var contentDbContext = InMemoryApplicationDbContext(contentDbContextId))
+            {
+                await contentDbContext.AddAsync(release);
+                await contentDbContext.SaveChangesAsync();
+            }
+
+            await using (var contentDbContext = InMemoryApplicationDbContext(contentDbContextId))
+            {
+                var userId = Guid.NewGuid();
+                var service = SetupContentService(
+                    contentDbContext: contentDbContext,
+                    userId: userId);
+                var result = await service.ResolveComment(comment.Id, true);
+
+                var viewModel = result.AssertRight();
+                Assert.Equal("Existing comment", viewModel.Content);
+                Assert.NotNull(viewModel.Resolved);
+
+                var comments = contentDbContext.Comment.ToList();
+                Assert.Single(comments);
+                Assert.Equal("Existing comment", comments[0].Content);
+                Assert.Equal(comment.CreatedById, comments[0].CreatedById);
+                Assert.Equal(userId, comments[0].ResolvedById);
+            }
+        }
+
+        [Fact]
+        public async Task ResolveComment_False()
+        {
+            var comment = new Comment
+            {
+                Content = "Existing comment",
+                CreatedById = Guid.NewGuid(),
+            };
+            var release = new Release
+            {
+                Content = new List<ReleaseContentSection>
+                {
+                    new ReleaseContentSection
+                    {
+                        ContentSection = new ContentSection
+                        {
+                            Content = new List<ContentBlock>
+                            {
+                                new HtmlBlock
+                                {
+                                    Comments = new List<Comment>
+                                    {
+                                        comment
```

Comment:
Probably good for this test to start off with a Resolved date and ResolvedBy and ResolvedById values, to prove that the method will unset them.